### DISCUSSION TITLE
grid mobile cols-span-NOTWORKING_PROPERLY

### DISCUSSION THREAD

**Discussion by nielsoffice** [tailwindlabs/tailwindcss#12334] [Help]
**What version of Tailwind CSS are you using?** 

For example: Latest version

**What build tool (or framework if it abstracts the build tool) are you using?**
  
For example: node just cli 

**What version of Node.js are you using?**

For example: v20.5.0

**What browser are you using?**

For example: Chrome

**What operating system are you using?**

For example: macOS, Windows

**Reproduction URL**

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**

we have screen ex sm, md and lg

```HTML
 <div class="grid grid-cols-12 ">
  <!-- means we should have 1 to 12 here --> 
  <div class="col-span-2"> here should I have 2 column occupied </div>
  <div class="col-span-10">here should I have 10 column occupied </div>
</div>

 <!-- but the result is  one cols occupied and one cols occupied  -->
 
```

```HTML
 <!-- NESTED ISSUE ! -->
 
 <div class="grid grid-cols-12 ">
  <!-- means we should have 1 to 12 here --> 
  <div class="col-span-2"> here should I have 2 column occupied </div>
  <div class="col-span-10 grid grid-cols-12"> <!-- I have "grid grid-cols-12" here -->
     here should I have 10 columns occupied 
     <!-- In children, I should have 12 cols available, right? but not in bootstrap that is work
      <div class="col-span-6"></div>
      <div class="col-span-6"></div>
   </div>
</div>


 
```

Describe the problem you're seeing, any important steps to reproduce and what behavior you expect instead.

**Comment by nielsoffice** [comment]
![image](https://github.com/tailwindlabs/tailwindcss/assets/67406593/ca233f03-c393-4c97-8caa-626f947d53c6)

**Comment by nielsoffice** [comment]
That should be 10 however it is 9?

**Comment by nielsoffice** [comment]
it's work in CDN anyway!

**Comment by adydetra** [comment]
I don't understand what you mean, try testing your code here.

https://play.tailwindcss.com/QAySVH9vjD
